sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/statefulOperators.scala

```diff
@@ -182,6 +191,27 @@ trait StateStoreWriter extends StatefulOperator with PythonSQLMetrics { self: Sp
     }
   }
 
+  val checkpointInfoAccumulator: CollectionAccumulator[StateStoreCheckpointInfo] = {
+    SparkContext.getActive.map(_.collectionAccumulator[StateStoreCheckpointInfo]).get
+  }
+
+  def getCheckpointInfo(): Array[StateStoreCheckpointInfo] = {
+    assert(conf.stateStoreCheckpointFormatVersion >= 2)
+    val ret = checkpointInfoAccumulator
+      .value
+      .asScala
+      .toSeq
+      .groupBy(_.partitionId)
+      .map {
+        case (key, values) => key -> values.head
+      }
```

Comment:
This list would be non-zero only if there was a task retry/speculative execution, right?

Reply:
And as discussed earlier today offline, this has the issue of not working if the same partition has multiple state stores, e.g. in a stream-stream join.

Reply:
> if there was a task retry/speculative execution

Also if there is a fan-out in foreachBatch, i.e. df.write.save() executed twice